### ISSUE TITLE
Add an id (`#`) field definition

### ISSUE THREAD

**Issue by joente** [thingsdb/ThingsDB#296]
**Is your feature request related to a problem? Please describe.**
Sometimes as a user, you want an alternative name for the id (`#`). For things this is not possible as we need a reserved key to avoid conflicts with other properties. For example, the text `id` would not work as this could be used for something else. However, on a Type it is possible as we control which fields are available.

**Describe the solution you'd like**
Add a `"#"` definition which can be used once on a type. For example:
```javascript
set_type('User', {
  id: '#',
  name: 'str'
});
.me = User{name: 'Iris'};
```
Would return something like, (the ID is just some example of course)
```json
{
  "id": 123,
  "name": "Iris"
}
```

**Describe alternatives you've considered**
As an alternative we could use the combination of the flag `NO_IDS` and a computed property in combination with _warp(..)_ but this seems to be an easy solution which also works when types are nested.

**Comment by joente** [comment]
We have to decide if we want this _id_ to be a real property, thus a property which can be used in ThingsDB code, _or_, do we use the _id_ property only on output.

I thought about the first option, to make it a real property. This however will affect both performance and the costs of a type while the Id is still accessible with the `.id()`.  I think it should be an output only property for this reason and that we don't need to introduce a new internal value with parent keeping etc.

**Comment by joente** [comment]
Implemented in **v1.3.0**

**Comment by joente** [comment]
See documentation:  https://docs.thingsdb.net/v1/overview/type/#named-id
